### DISCUSSION TITLE
dokku config:set complains about App name must begin with lowercase alphanumeric character

### DISCUSSION THREAD

**Discussion by justgage** [dokku/dokku#4591] [Q&A] [answered]
So I just started using dokku today and I ran into the following error trying to set config variables:

```
ubuntu# dokku config:set --no-restart PORT=4000
 !     App name must begin with lowercase alphanumeric character, and cannot include uppercase characters, colons, or u
 ```
 
 As far as I can tell my app doesn't have uppercase letters:
 
 ```
ubuntu# dokku apps:list
=====> My Apps
topical
 ```
 
 Sorry, this is probably a dumb error, please help 🙏🏻

**Comment by justgage** [comment]
Looking through the docs I think I was referencing an older tutorial the new command is this:

```
dokku config:set appname PORT=4000
```

note replace `appname` with your app's name

**Comment by Cellane** [comment, marked as answer]
I’m glad you found the answer, but just to clarify, an app name is almost always required, you need to specify which app’s environmental variables you want to set/change. The only exception would be `dokku config:set --global` argument, which sets the key-value pair to all apps.

Finally, the `--no-restart` argument is still supported, and must precede the app name, like so: `dokku config:set --no-restart appname PORT=4000`. For more information, check `dokku config:help`.

**Reply by justgage** [reply]
Thanks for the added detail :)
